File: TCP_Python/TCP_Client.py
from socket import *
serverName = 'localhost'     # serverName = 'localhost' or private IP address of the server
serverPort = 12000


clientSocket = socket(AF_INET, SOCK_STREAM)
clientSocket.connect((serverName, serverPort))

message = input('Input lowercase sentence:')
                        
clientSocket.send(message.encode())

modifiedSentence = clientSocket.recv(1024)

print('Reply From Server: ', modifiedSentence.decode())


clientSocket.close()


# # run continlusly chat application

# while True:
#     clientSocket = socket(AF_INET, SOCK_STREAM)
#     clientSocket.connect((serverName, serverPort))

#     message = input('Input lowercase sentence:')
                        
#     clientSocket.send(message.encode())

#     modifiedSentence = clientSocket.recv(1024)

#     print('Reply From Server: ', modifiedSentence.decode())


#     clientSocket.close()

File: TCP_C++/TCP_Client.cpp
#include <iostream>
#include <cstdlib>
#include <cstring>
#include <unistd.h>
#include <arpa/inet.h>

#define PORT 12000
#define SERVER_IP "127.0.0.1"
#define BUFFER_SIZE 1024

int main() {
    int sock;
    struct sockaddr_in server_addr;
    char buffer[BUFFER_SIZE] = {0};

    // Create socket
    if ((sock = socket(AF_INET, SOCK_STREAM, 0)) < 0) {
        perror("Socket failed");
        exit(EXIT_FAILURE);
    }

    server_addr.sin_family = AF_INET;
    server_addr.sin_port = htons(PORT);


    if (inet_pton(AF_INET, SERVER_IP, &server_addr.sin_addr) <= 0) {
        perror("Invalid address");
        exit(EXIT_FAILURE);
    }

    // Connect to the server
    if (connect(sock, (struct sockaddr*)&server_addr, sizeof(server_addr)) < 0) {
        perror("Connection failed");
        exit(EXIT_FAILURE);
    }

    std::cout << "Connected to server!" << std::endl;

    // Send and receive 
    while (true) {
        std::cout << "Enter message: ";
        std::cin.getline(buffer, BUFFER_SIZE);

        send(sock, buffer, strlen(buffer), 0);
        int bytes_received = recv(sock, buffer, BUFFER_SIZE, 0);

        if (bytes_received <= 0) {
            std::cout << "Server closed the connection." << std::endl;
            break;
        }

        buffer[bytes_received] = '\0';
        std::cout << "Server: " << buffer << std::endl;
    }


    close(sock);

    return 0;
}
